src/main/authentication/authentication.ts

```diff
@@ -23,10 +25,23 @@ async function piAadVerifyFunction(iss, sub, profile, accessToken, refreshToken,
     }
 }
 
-async function serializeUser(foundUser, done) {
-    if (foundUser['flow'] === 'CFT') {
-        const user = await accountManagementRequests.getPiUserByCftID(foundUser['uid']);
+async function ssoVerifyFunction(iss, sub, profile, accessToken, refreshToken, done): Promise<any> {
+    const userRole = await ssoAuthentication.determineUserRole(profile.oid, accessToken);
+    if (userRole) {
+        profile['roles'] = userRole;
+        profile['email'] = profile._json['preferred_username'];
+        profile['flow'] = 'SSO';
+        const response = await ssoAuthentication.handleSsoUser(profile);
```

Comment:
For CFT IDAM. the creation is done in the 'SerializeUser' function instead so I think it makes sense to keep them consistent

Reply:
The user creation needs to be done here for SSO for error handling. If we fail to create the system admin or admin account (particularly the system admin as it can fail when hitting the max size), I am setting the req.user.created to false, so it can render the appropriate error page in processSsoSignIn().
It was originally done in serializeUser() but req.user cannot be set there so I moved it here.